README.md

```diff
@@ -237,6 +237,180 @@ You can do the same for Kafka Streams and Kafka, by using the `kstreams` and `ka
 
 Lastly, create a Makefile in the `code` directory to invoke the harness runner and check any outputs that it produces. Then modify the `.semaphore/semaphore.yml` file to invoke that Makefile. This will make sure your tutorial gets checked by the CI system.
 
+## Working with the harness file
+
+Given the test harness is the `heart` of a tutorial, it will be helpful to describe in detail how to work with a `kafka|ksql|kstreams.yml` file.  It should be noted the harness file is in the [YAML file formt](https://en.wikipedia.org/wiki/YAML), so formatting properly is essential.  The harness files generates the structure of the rendered tutorial and also validates any output of tutorial steps against expected values.
```

Comment:
@bbejeck can we had a Table of Contents at the top of the README?

Reply:
@ybyzek I took a stab at this, check it out and let me know WYT.

Reply:
@bbejeck Awesome!  For your consideration: decrease it 2 levels instead of 3 levels -- makes it easier for future maintenance so you're not chasing down changes at the 3rd level back to the TOC.  But if you prefer 3, keep it ;). Looks great!

Reply:
@ybyzek great idea, will do